### PR TITLE
Micronaut Http Server exposed as Functions

### PR DESCRIPTION
**Pull request by driverpt**
Attempts to address #8661

### CODE REVIEW

**Comment by driverpt** [comment]
@sdelamo , can you let me know once RouteExecutor is stable ? So i can resume this PR

File: http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java [hunk 1]
```diff
@@ -524,6 +524,10 @@ ExecutionFlow<MutableHttpResponse<?>> createResponseForBody(HttpRequest<?> reque
                 }
                 response.body(null);
             }
+            if (response.getContentType().isEmpty()) {
+                MediaType mediaType = resolveDefaultResponseContentType(request, routeInfo);
+                response.contentType(mediaType);
+            }
```
Comment (graemerocher) [inline http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java:530]: not sure about this change, why is this necessary?

**Comment by graemerocher** [comment]
Seems to be missing a `build.gradle` definition. What is the purpose of this implementation?

File: http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java (hunk 1 above)
Comment (driverpt) [reply http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java:530]: Hello @graemerocher , according to testing, if this is not set, no `Content-Type` is set on response

**Comment by driverpt** [comment]
@graemerocher , the purpose of this would be to expose Micronaut Http as a Function (similar to what exists in Go). The purpose would be to facilitate the Integration of Micronaut with Function-based Frameworks (AWS Lambda, Google Cloud Function, Azure Function, etc...). I have created an example in [this PR](https://github.com/micronaut-projects/micronaut-aws/pull/1595/files#diff-7b4dacb23e848467cdf4a8f6badd2f4a6e720526670e05f52da8a6180240c94fR52-R64)

File: http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java (hunk 1 above)
Comment (graemerocher) [reply http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java:530]: don't think the core Router should be changed for this, btw there are already implementation of the routing for serverless that are based on the Servlet module like https://github.com/micronaut-projects/micronaut-gcp/tree/master/gcp-function-http

**Comment by graemerocher** [comment]
I understand that but we already have integration with Lambda and GCP that use the Servlet module, not sure of the need for a new one? https://github.com/micronaut-projects/micronaut-gcp/tree/master/gcp-function-http

**Comment by driverpt** [comment]
This is what i'm trying to achieve: https://github.com/awslabs/aws-lambda-go-api-proxy/blob/master/echo/adapterv2.go .

**Comment by driverpt** [comment]
> I understand that but we already have integration with Lambda and GCP that use the Servlet module, not sure of the need for a new one? https://github.com/micronaut-projects/micronaut-gcp/tree/master/gcp-function-http

AFAIK, AWS Lambda doesn't use Servlet

**Comment by graemerocher** [comment]
no it uses aws serverless container underneath. Is there something deficient with this module that means we need 2 ways?

**Comment by driverpt** [comment]
With all due respect to the maintainers, AWS Serverless Container brings more issues than the ones it solves.

AWS has 3 types of payloads (Payload V1, Payload V2 and ALB Payload, the latter is almost equal to V2) for Http Serving Lambdas. 2 of them are similar, but from Payload V1 to V2 there are major differences. IMHO this can be easily solved by abstracting the input to Micronaut and just implement an HttpRequest mapping layer.
And I'm pretty sure that in the future there will be more Payload types.

**Comment by driverpt** [comment]
@graemerocher @sdelamo , just to clarify. Should I cancel this PR as it will not be implemented or can I keep it open?

**Comment by sdelamo** [comment]
I agree with you, AWS Serverless container has served as well but it  is problematic. I think with the server tck we are in a position to get rid of its dependency. We are trying to solve it in the AWS module https://github.com/micronaut-projects/micronaut-aws/pull/1648

**Comment by driverpt** [comment]
I've added some comments about the discoveries I've made. Let me know if I can further help you.

**Comment by sdelamo** [comment]
Closing this for now as I think we have a good support in Lambda for the three kinds of payloads Payload V1, Payload V2 and ALB Payload without the use of  AWS Serverless Container.
